File: scala/finatra-demo/src/main/scala/com/example/jpa/Role.scala
package com.example.jpa

import javax.persistence.{Column, _}

import scala.beans.BeanProperty
import scala.collection.mutable.ListBuffer

@Entity
@Table(name = "t_role")
class Role extends BaseEntity {
  @BeanProperty
  @Column(name = "role_name", nullable = false, unique = true)
  var roleName: String = _

  @BeanProperty
  @ManyToMany(fetch = FetchType.EAGER, mappedBy = "roles")
  var users: ListBuffer[LoginUser] = _
}

object Role {
  def apply(roleName: String): Role = {
    val role = new Role()
    role.roleName = roleName
    role
  }
}

File: scala/finatra-demo/src/main/java/com/example/jpa/Role.java
package com.example.jpa;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "t_role")
public class Role extends BaseEntity {
    @Column(name = "role_name", nullable = false, unique = true)
    public String roleName;
    @ManyToMany(fetch = FetchType.EAGER, mappedBy = "roles")
    //@ElementCollection(targetClass = classOf[LoginUser])
    public List<LoginUser> users;

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }

    public List<LoginUser> getUsers() {
        return users;
    }

    public void setUsers(List<LoginUser> users) {
        this.users = users;
    }
}
